modin/pandas/test/dataframe/test_indexing.py

```diff
@@ -2117,3 +2117,22 @@ def test__getitem_bool_single_row_dataframe():
     # This test case comes from
     # https://github.com/modin-project/modin/issues/4845
     eval_general(pd, pandas, lambda lib: lib.DataFrame([1])[lib.Series([True])])
+
+
+# This is a very subtle bug that comes from:
+# https://github.com/modin-project/modin/issues/4945
+def test_4945():
```

Comment:
Maybe we can give this test a different name now? Can't think of a good one off the top of my head. Maybe something like `test_lazy_eval_index` ?

Reply:
Agree